packages/mobile/src/components/details-tile/DetailsTile.tsx

```diff
@@ -179,8 +188,16 @@ export const DetailsTile = ({
   saveCount,
   headerText,
   title,
-  user
+  user,
+  track
 }: DetailsTileProps) => {
+  const isPremiumContentEnabled = useIsPremiumContentEnabled()
+  const { doesUserHaveAccess } = usePremiumContentAccess(
+    track ? (track as unknown as Track) : null
```

Comment:
Why do we need the cast here?

Reply:
because it could alse be a `SearchTrack`, but we only care about certain fields, which both types have